pennylane_lightning/lightning_qubit/lightning_qubit.py

```diff
@@ -75,6 +75,11 @@
         VectorJacobianProductC128,
     )
 
+    def _state_dtype(dtype):
+        if dtype not in [np.complex128, np.complex64]:  # pragma: no cover
+            raise ValueError(f"Data type is not supported for state-vector computation: {dtype}")
+        return StateVectorC128 if dtype == np.complex128 else StateVectorC64
```

Comment:
Do we need a test to silence the below warning?

Reply:
This warning appeared because one of the upload code coverage steps failed. Coverage should be OK, as we can see in [e800725](https://github.com/PennyLaneAI/pennylane-lightning/pull/601/commits/e80072542253866431dcf9286834335666e20b4e).